.github/workflows/scripts/pre_before_install.sh

```diff
@@ -1,18 +1,24 @@
 set -mveuo pipefail
 
-
-update_docker_configuration() {
+if [ -f "/etc/docker/daemon.json" ]
+then
   echo "INFO:
   Updating docker configuration
   "
 
-  echo '{
-  "registry-mirrors": ["https://mirror.gcr.io"],
-  "insecure-registries" : ["pulp:80"],
-  "mtu": 1460
-}' | sudo tee /etc/docker/daemon.json
-  sudo service docker restart
-}
+  echo "$(cat /etc/docker/daemon.json | jq -s '.[0] + {
+  "insecure-registries" : ["pulp:80", "pulp.example.com:80", "pulp.example.com"]
```

Comment:
GHA comes with podman, pulp-smash uses podman when it is installed, so you also need to update podman:
```
update_podman_configuration() {
  echo "INFO:
  Updating podman configuration
  "
  echo "[registries.insecure]" | sudo tee -a /etc/containers/registries.conf
  echo "registries = ['pulp.example.com:80']" | sudo tee -a /etc/containers/registries.conf
  podman info
}
```

Reply:
Fair enough, i will leave only podman config update.
I think on travis podman was not available so we used docker.

Reply:
actually the reason i updated this code to check what is available docker or podman is because it took me quite a lot of time to debug why client pull failed, and the reason was that in travis we used docker and when we moved to gha it does not have docker but podman